Aplib.Core/Belief/Beliefs/IBelief.cs

```diff
@@ -3,7 +3,7 @@
     /// <summary>
     /// A belief represents/encapsulates an observation (i.e., piece of information of the game state as perceived by an agent).
     /// </summary>
-    public interface IBelief
+    public interface IBelief : IDocumented
```

Comment:
I would advise against letting an interface implement another, unless both are concept-wise tightly bound together.

Otherwise, if someone were to implement his own `IBelief` for testing purposes, for example, he would be required to implement `IDocumented` as well.

In other words, it is not _always_ the case that, inheritly, a belief has documentation coming with it.

Reply:
For this reason, I let every implementation of `IBelief` inherit `IDocumented`.

If you somehow require that you can get an `IMetadata` from an `IBelief`, then I would suggest doing something like this:

```csharp
if (belief is IDocumented beliefDocumentation)
  var docs = beliefDocumentation;
```

or

```csharp
IDocumented? docs = belief as IDocumented; // Returns null if it does not implement IDocumented
```